Bio/motifs/__init__.py

```diff
@@ -160,8 +160,12 @@ class Instances(list):
 
     def __init__(self, instances=None, alphabet=None):
         """Initialize the class."""
-        from Bio.Alphabet import IUPAC
         from Bio.Seq import Seq
+        try:
+            # Received an old-style alphabet
+            alphabet = alphabet.letters
```

Comment:
Could you call the internal variable letters instead of alphabet here? I would find that clearer.